### PR TITLE
LibUnicode: Upgrade to Unicode version 14.0.0

### PR DESCRIPTION
**Pull request by trflynn89**
This functionalizes some of the work in `unicode_data.cmake` for clarity and adds support for re-downloading the UCD and CLDR when their versions change. Then changes the UCD version to 14.0.0 :)

### CODE REVIEW

**Comment by trflynn89** [comment]
Marking as a draft because this will cause test262 failures until https://github.com/tc39/test262/pull/3199 is merged.

File: Meta/CMake/unicode_data.cmake [hunk 1]
```diff
@@ -1,139 +1,124 @@
-set(LOCALE_DATA_CACHE_LOCATION "${CMAKE_BINARY_DIR}/CLDR" CACHE PATH "Download location for CLDR files")
-set(UNICODE_CACHE_LOCATION "${CMAKE_BINARY_DIR}/UCD" CACHE PATH "Download location for UCD files")
+set(UCD_VERSION 14.0.0)
+set(CLDR_VERSION 39.0.0)
 
-set(UNICODE_DATA_URL https://www.unicode.org/Public/13.0.0/ucd/UnicodeData.txt)
-set(UNICODE_DATA_PATH ${UNICODE_CACHE_LOCATION}/UnicodeData.txt)
+set(UCD_PATH "${CMAKE_BINARY_DIR}/UCD" CACHE PATH "Download location for UCD files")
+set(CLDR_PATH "${CMAKE_BINARY_DIR}/CLDR" CACHE PATH "Download location for CLDR files")
```
Comment (ADKaster) [inline Meta/CMake/unicode_data.cmake:5]: Fwiw the CI scripts and Superbuild.cmake were setting the path for these cached artifacts using the old name (for when you get back to this :) )

File: Meta/CMake/unicode_data.cmake [hunk 2]
```diff
@@ -1,139 +1,124 @@
...
+        file(STRINGS ${version_file} active_version)
+        if (version STREQUAL active_version)
+            set(version_differs NO)
+        endif()
     endif()
-    if (NOT EXISTS ${SCRIPT_EXTENSIONS_PATH})
-        message(STATUS "Downloading UCD ScriptExtensions.txt from ${SCRIPT_EXTENSIONS_URL}...")
-        file(DOWNLOAD ${SCRIPT_EXTENSIONS_URL} ${SCRIPT_EXTENSIONS_PATH} INACTIVITY_TIMEOUT 10)
+
+    if (${version_differs})
```
Comment (friendlyanon) [inline Meta/CMake/unicode_data.cmake:71]: Prefer to just name the variable alone if the [documentation](https://cmake.org/cmake/help/latest/command/if.html) indicates `<variable|string>`:

```suggestion
    if (version_differs)
```

An unquoted variable substitution (aka list expansion) into an `if()` command *can* do some unintended things.

**Review by friendlyanon — CHANGES_REQUESTED** [review]
As a general rule when touching CMake code, please consider quoting every variable substitution *unless* it's intended to be a list expansion.

**Comment by trflynn89** [comment]
This should be good to go now, the test262 PR has been merged.

**Pull request merged**